packages/transaction-preview/src/views/Index.svelte

```diff
@@ -0,0 +1,47 @@
+<script lang="ts">
+  import { onMount } from 'svelte'
+  import { fade } from 'svelte/transition'
+  import { cubicOut } from 'svelte/easing'
+  import type { SimPlatformResponse } from '../types'
+  import Maximized from './Maximized.svelte'
+  import Minimized from './Minimized.svelte'
+  import { getDevice } from '../utils'
+  import type { Subject } from 'rxjs'
+
+  export let simResponse: SimPlatformResponse
+  export let requireTransactionApproval: boolean
+  export let approved$: Subject<boolean>
+  export let destroyApp: () => void
+
+  $: expanded = device.type === 'mobile' ? false : true
+
+  const device = getDevice()
+  const startTime = Date.now()
+
+  function toggleExpanded(maximize: boolean) {
+    expanded = maximize
+  }
+
+  const transactionApproved = (approved: boolean) => {
+    approved$.next(approved)
+  }
+  let mounted = false
+  onMount(() => (mounted = true))
+</script>
+
+{#if mounted}
```

Comment:
Curious why this conditional was needed? Svelte should handle all of this internally. Or is this to ensure animation on first mount? If so you can add the [`intro` option](https://svelte.dev/docs#run-time-client-side-component-api) when creating the component.

Reply:
Yep that was the reasoning, thank you for this! Way cleaner!